test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageExtractorTests.cs

```diff
@@ -1889,6 +1890,225 @@ public async Task GetTrustResultAsync_RepositoryPrimarySignedPackage_PackageSign
+                        test.Source,
+                        packageReader,
+                        test.Resolver,
+                        test.Context,
+                        CancellationToken.None);
+                }
+                catch (SignatureException e)
+                {
+                    exception = e;
+                }
+
+                // Assert
+                exception.Should().NotBeNull();
+            }
+        }
+
+        [Fact]
+        public async Task ExtractPackageAsync_RequireMode_UnsignedPackage_PackageFolderReader_WhenUnsignedPackagesDisallowed_SkipsSigningVerificationAsync()
+        {
+            // Arrange
+            var signedPackageVerifier = new Mock<IPackageSignatureVerifier>(MockBehavior.Strict);
+
+            signedPackageVerifier.Setup(x => x.VerifySignaturesAsync(
+                It.IsAny<ISignedPackageReader>(),
+                It.IsAny<SignedPackageVerifierSettings>(),
+                It.IsAny<CancellationToken>(),
+                It.IsAny<Guid>())).
+                ReturnsAsync(new VerifySignaturesResult(valid: false, signed: false));
+
+            var extractionContext = new PackageExtractionContext(
+                packageSaveMode: PackageSaveMode.Nuspec | PackageSaveMode.Files,
+                xmlDocFileSaveMode: XmlDocFileSaveMode.None,
+                clientPolicyContext: new ClientPolicyContext(SignatureValidationMode.Require, allowList: null),
+                logger: NullLogger.Instance)
+            {
+                SignedPackageVerifier = signedPackageVerifier.Object
+            };
+
+            using (var test = new ExtractPackageAsyncTest(extractionContext))
+            using (var packageDir = TestDirectory.Create())
+            {
+                var packageContext = new SimpleTestPackageContext();
+                await SimpleTestPackageUtility.CreatePackagesAsync(test.Source, packageContext);
+
+                var packageFile = new FileInfo(Path.Combine(test.Source,
+                    $"{packageContext.Identity.Id}.{packageContext.Identity.Version.ToNormalizedString()}.nupkg"));
+
+                using (var zipFile = new ZipArchive(File.OpenRead(packageFile.FullName)))
+                {
+                    zipFile.ExtractAll(packageDir.Path);
+                }
+
+                var packageReader = new PackageFolderReader(packageDir);
+
+                // Act
+                SignatureException exception = null;
+
+                try
+                {
+                    await PackageExtractor.ExtractPackageAsync(
+                        test.Source,
+                        packageReader,
+                        test.Resolver,
+                        test.Context,
+                        CancellationToken.None);
+                }
+                catch (SignatureException e)
+                {
+                    exception = e;
+                }
+
+                // Assert
+                exception.Should().BeNull();
```

Comment:
Should you verify the package was actually extracted?

Reply:
do you think it would be sufficient to verify that the result from `ExtractPackageAsync(…)` is not empty? (It returns a list of filepaths for all the extracted files)

Reply:
It's better than nothing.